File: csharp/Day1/Program.cs
using System;
using System.IO;
using System.Text;
using System.Linq;

var data = File.ReadAllText("input.txt");
var query =
    from calorieList in data.Split($"{Environment.NewLine}{Environment.NewLine}")
    let caloriePerElf = calorieList.Split(Environment.NewLine).Select(int.Parse).Sum()
    orderby caloriePerElf descending
    select caloriePerElf;

var answerPartOne = query.First();
Console.WriteLine($"Answer Part One: {answerPartOne}");

var answerPartTwo = query.Take(3).Sum();
Console.WriteLine($"Answer Part Two: {answerPartTwo}");

File: fsharp/Day1/Program.fs
open System
open System.IO
open System.Text.RegularExpressions

let SplitString delimiter (str: string) =
    let regex = new Regex(delimiter)
    regex.Split(str)

let input = "input.txt" |> File.ReadAllText

let calculateCalories =
    SplitString Environment.NewLine >> Array.map int >> Array.sum

let caloriesPerElf =
    input
    |> SplitString(Environment.NewLine + Environment.NewLine)
    |> Array.map calculateCalories

let answerPartOne = caloriesPerElf |> Array.max

let answerPartTwo =
    caloriesPerElf |> Array.sortDescending |> Array.take 3 |> Array.sum

printf "Answer part one: %i%s" answerPartOne Environment.NewLine
printf "Answer part two: %i%s" answerPartTwo Environment.NewLine
